common/cloud_builder.py

```diff
@@ -0,0 +1,258 @@
+"""CloudBuilder executes agents in Cloud Builds."""
+import argparse
+import logging
+import os
+import subprocess
+import time
+import uuid
+from typing import Any
+
+import google.api_core.client_options
+import googleapiclient.errors
+from google.api_core.exceptions import NotFound
+from google.auth import default
+from google.cloud import storage
+from google.oauth2.service_account import Credentials
+from googleapiclient.discovery import build as cloud_build
+
+import utils
+from agent.base_agent import BaseAgent
+from results import BuildResult, Result
+
+OFG_REPO = 'https://github.com/google/oss-fuzz-gen.git'
+OF_REPO = 'https://github.com/google/oss-fuzz.git'
+US_CENTRAL_CLIENT_OPTIONS = google.api_core.client_options.ClientOptions(
+    api_endpoint='https://us-central1-cloudbuild.googleapis.com/')
+
+
+class CloudBuilder:
+  """Encapsulate functions to request Google Cloud Builds to execute agents."""
+
+  def __init__(self, args: argparse.Namespace) -> None:
+    #TODO(dongge): extra tags.
+    _, self.project_id = default()
+    assert self.project_id, 'Cloud experiment requires a Google cloud project.'
+
+    self.bucket_name = args.cloud_experiment_bucket
+    self.bucket = storage.Client().bucket(self.bucket_name)
+
+    # Service account.
+    self.service_account = os.getenv('GOOGLE_SERVICE_ACCOUNT', '')
+    service_account_file = os.getenv('GOOGLE_APPLICATION_CREDENTIALS', '')
```

Comment:
Do we need this? Can we just use `google.auth.default()` ? https://google-auth.readthedocs.io/en/master/reference/google.auth.html#google.auth.default

This gets the most suitable credentials in each environment. This could be a service account key, a service account from GCE, or local gcloud creds.

Reply:
Done!